src/charmed_kubeflow_chisme/components/component_graph.py

```diff
@@ -23,6 +23,10 @@ def __init__(self):
         self.component_items: dict[str, ComponentGraphItem] = {}
         self.status_prioritiser = Prioritiser()
 
+    def __len__(self) -> int:
+        """Returns the number of component_items we have."""
+        return len(self.component_items)
+
```

Comment:
why do we go with a `len()` instead of sth like self._component_graph.length()?

Reply:
I think this is a python convention.  

functionally, doing `self._component_graph.length()` here would definitely work.  But python data types typically support this through the `__len__` dunder method so they can be used similar to the builtin types.  Best explanations I can find are [here](https://lucumr.pocoo.org/2011/7/9/python-and-pola/), especially the "Why are you Astonished" section (lol), and [here](https://docs.python.org/3/reference/datamodel.html#emulating-container-types) in the docs.

But, if you object after reading those, we should discuss! :)

Reply:
Sounds good, thanks for the context. The article was a good read.